mitodl/micromasters
# refactor: remove the usages of old terms of service and use CMS based page

#### Pre-Flight checklist


#### What are the relevant tickets?
#5102 

#### What's this PR do?
- Removes the old terms of service page
- Refactors all the usages of the old independent refactor page
- Fixed/removes irrelevant tests
- support CMS based terms of service page

#### How should this be manually tested?
- Run the application & open CMS
- Create a resource page ins CMS and set its slug to `terms_of_service` in the `Promote tab`
-  Check the terms of service page from the `Footer, SignIn, Create Account, Order Summary` (It should load the new tos page created in CMS and there should be no errors regarding tos in the console)

**Reviewer Note:**
- The TOS page link from anywhere would open in a new tab (e.g. Sign In, Crate Account, Order Summary, Footer)
- Once you create a Resource page in CMS (Make sure you set `terms_of_service` in promote tab)
- Open the TOS link from any page as mentioned in point#1
- It's currently unsure either to use `terms_of_service` or `terms-of-service` as the URL path for TOS

#### Screenshots (if appropriate)
**OLD TOS PAGE**
![screencapture-micromasters-rc-odl-mit-edu-terms-of-service-2021-10-20-17_44_18](https://user-images.githubusercontent.com/34372316/138098286-9e3bab7c-2313-42d0-8a13-2518bc5a7e83.png)


**NEW TOS PAGE(Through Resource Pages) -- Desktop**
![screencapture-mm-odl-local-8079-terms-of-service-2021-10-20-18_02_47](https://user-images.githubusercontent.com/34372316/138098304-d6a76691-1d0e-466d-805a-9f4e0b48b3d0.png)

**NEW TOS PAGE(Through Resource Pages)  -- MOBILE**

![screencapture-mm-odl-local-8079-terms-of-service-2021-10-20-18_03_18](https://user-images.githubusercontent.com/34372316/138098314-27ee4c9e-8af6-4c26-af17-e34f8e6c6763.png)
